cdap-app-fabric/src/main/java/co/cask/cdap/app/DefaultAppConfigurer.java

```diff
@@ -193,10 +194,17 @@ public void addSchedule(Schedule schedule, SchedulableProgramType programType, S
                                   "Schedule data trigger must be greater than 0.");
     }
 
+    // TODO: [CDAP-11575] Temporary solution before REST API is merged. ScheduleSpecification will be removed and
+    // the block of code below will be refactored
     ScheduleSpecification spec =
       new ScheduleSpecification(schedule, new ScheduleProgramInfo(programType, programName), properties);
 
     schedules.put(schedule.getName(), spec);
+    ScheduleCreationSpec creationSpec = Schedulers.toScheduleCreationSpec(spec, deployNamespace.toEntityId());
+    Preconditions.checkArgument(null == programSchedules.put(schedule.getName(), creationSpec),
```

Comment:
it seems that for a usr, it would be a lot clearer to say: Duplicate schedule name for schedules: <existing one>, <new one>